TTC.Deployment.Tests/StackParameterReaderTest.cs

```diff
@@ -0,0 +1,144 @@
+
+                Assert.IsNotNull(parameters);
+                Assert.AreEqual(0, parameters.Count);
+            }
+            finally
+            {
+                DeleteFile(filePath);
+            }
+        }
+
+        [Test]
+        public void Read_WhenEmptyJson_ReturnsEmptyList()
+        {
+            var filePath = "ReadsOneParameter.txt";
+            var content = "{}";
+
+            CreateFile(filePath, content);
+
+            try
+            {
+                var reader = new StackParameterReader(filePath);
+                List<Parameter> parameters = reader.Read();
+
+                Assert.IsNotNull(parameters);
+                Assert.AreEqual(0, parameters.Count);
+            }
+            finally
+            {
+                DeleteFile(filePath);
+            }
+        }
+
+        [Test]
+        public void Read_WhenInvalidJson_Throws()
+        {
+            var filePath = "ReadsOneParameter.txt";
+            var content = "{abc}";
+
+            CreateFile(filePath, content);
+
+            try
+            {
+                var reader = new StackParameterReader(filePath);
+
+                Assert.Throws<JsonReaderException>(() => reader.Read());
+            }
+            finally
+            {
+                DeleteFile(filePath);
+            }
+        }
+
+        [Test]
+        public void Read_WhenOneParameter_ReturnsIt()
+        {
+            var filePath = "ReadsOneParameter.txt";
+            var content = @"{""key1"" : ""value1""}";
+
+            CreateFile(filePath, content);
+
+            try
+            {
+                var reader = new StackParameterReader(filePath);
+                List<Parameter> parameters = reader.Read();
+
+                Assert.AreEqual(1, parameters.Count);
+                Assert.AreEqual("key1", parameters[0].ParameterKey);
+                Assert.AreEqual("value1", parameters[0].ParameterValue);
+            }
+            finally
+            {
+                DeleteFile(filePath);
+            }
+        }
+
+        [Test]
+        public void Read_WhenMultipmePatameters_ReturnsThemAll()
+        {
+            var filePath = "ReadsOneParameter.txt";
+            var content =
+                @"{
+                    ""key1"" : ""value1"",
+                    ""key2"" : ""value2""
+                }";
+
+            CreateFile(filePath, content);
+
+            try
+            {
+                var reader = new StackParameterReader(filePath);
+                List<Parameter> parameters = reader.Read();
+
+                Assert.AreEqual(2, parameters.Count);
+                Assert.AreEqual("key1", parameters[0].ParameterKey);
+                Assert.AreEqual("value1", parameters[0].ParameterValue);
+                Assert.AreEqual("key2", parameters[1].ParameterKey);
+                Assert.AreEqual("value2", parameters[1].ParameterValue);
+            }
+            finally
+            {
+                DeleteFile(filePath);
```

Comment:
We do this in each test, and `filePath` never changes... that means we could remove the try/finally from each test and move `DeleteFile(filePath)` into a TearDown method instead... if you want :wink: